### ISSUE TITLE
Error removing slides

### ISSUE THREAD

**Issue by sergio-poliedro** [ShapeCrawler/ShapeCrawler#749]
Hello! I have the following code that was working with version 0.45:

`pres.Slides.Remove(pres.Slides[1]);`
`pres.Slides.Remove(pres.Slides[1]);`
`pres.Slides.Remove(pres.Slides[1]);`

When I updated to last version (0.54) it stopped working. In the second instruction I get an error:

`System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'id')
   at DocumentFormat.OpenXml.Packaging.OpenXmlPartContainer.GetPartById(String id)`

If I only remove the first slide, the PPT opens with a "repair PPT" prompt and the slide is not removed without any content.

Thanks in advance

**Comment by ashahabov** [comment]
Hello @sergio-poliedro,

I cannot reproduce this issue, for example, with the following code:

```C#
var pres = new Presentation("pres-with-4-slides.pptx");

pres.Slides.Remove(pres.Slides[1]);
pres.Slides.Remove(pres.Slides[1]);
pres.Slides.Remove(pres.Slides[1]);
```

Could you please share a more detail. It would also be good to attach a presentation sample.

**Comment by sergio-poliedro** [comment]
This is a larger version of the code:

```c#
var path = $"{Directory.GetCurrentDirectory()}/wwwroot/ppts/PPT_LocationSummary.pptx"; 

var pres = new Presentation(path);
var shapeCollection = pres.Slides[0].Shapes;

var autoShape = shapeCollection.GetById<IShape>(2);
autoShape.TextFrame!.Text = "Name";
autoShape = shapeCollection.GetById<IShape>(3);

autoShape.TextFrame!.Text = DateTime.Now.ToString("MMMM yyyy").ToUpper();

int totalSlides = 3;

pres.Slides[3].Number = totalSlides; 
pres.Slides.Remove(pres.Slides[1]); 
pres.Slides.Remove(pres.Slides[1]); 
``` 
And here is the PPT that I open: [PPT_LocationSummary.pptx](https://github.com/user-attachments/files/16821504/PPT_LocationSummary.pptx)

**Comment by ashahabov** [comment]
This also works for me😕

Do you use the latest 0.54.1 version?

**Comment by sergio-poliedro** [comment]
Yes, I'm using the latest 0.54.1 version. I've tried everything, I've also downgraded and I get this issue since version 0.50.

After I delete the first slide, the pres.Slides doesn't get accessible. I cannot even get the slides count:

```c#
pres.Slides.Remove(pres.Slides[1]);
var newTotalSlides = pres.Slides.Count;
```

**Comment by ashahabov** [comment]
What is the OS and Target Framework of environment?

**Comment by sergio-poliedro** [comment]
Windows 11, .Net 8.0

**Comment by ashahabov** [comment]
Hmm, I have the same env.

Then I would suggest you create a small console app that reproduces this issue and shares it here via .zip archive.

**Comment by sergio-poliedro** [comment]
I just created the following console app and still receiving the same error:

[Console app](https://we.tl/t-mLfGpOn0w2)

**Comment by ashahabov** [comment]
Now I can reproduce it!

So far I think the problem is that Application is using a different version of `DocumentFormat.OpenXml` than the one in ShapeCrawler.

![image](https://github.com/user-attachments/assets/b37ff5e9-0f45-46f3-827a-971032909f7a)

I tried to upgrade `DocumentFormat.OpenXml` in the ShapeCrawler too, but there were some errors. Need to try again.

@sergio-poliedro, thank you for the bug report.

**Comment by sergio-poliedro** [comment]
Great! Thanks for the help. I was able to bypass it downgrading DocumentFormat.OpenXml to version 3.0.2
